backends/gstreamer/source-element/src/source.rs

```diff
@@ -123,6 +127,36 @@ mod imp {
 
             let ghost_pad = gst::GhostPad::new("src", &pad)
                 .expect("Could not create src ghost pad");
+
+            // In order to make buffering/downloading work as we want, apart from
+            // setting the appropriate flags on the player playbin,
+            // the source needs to either:
+            //
+            // 1. be an http, mms, etc. scheme
+            // 2. report that it is "bandwidth limited".
+            //
+            // 1. is not straightforward because we are using a servosrc scheme for now.
+            // This may change in the future if we end up handling http/https/data
+            // URIs, which is what WebKit does.
+            //
+            // For 2. we need to make servosrc handle the scheduling properties query
+            // to report that it "is bandwidth limited".
+            ghost_pad.add_probe(gst::PadProbeType::QUERY_UPSTREAM, |_, probe_info| {
+                if let Some(gst::PadProbeData::Query(ref mut query)) = probe_info.data {
+                    if let QueryView::Scheduling(ref mut query) = query.view_mut() {
+                        query.set(
+                            gst::SchedulingFlags::SEQUENTIAL |
+                            gst::SchedulingFlags::BANDWIDTH_LIMITED,
+                            1, -1, 0
+                        );
+                        query.add_scheduling_modes(&[gst::PadMode::Push]);
+                        println!("QUERY {:?}", query);
+                        return gst::PadProbeReturn::Handled;
+                    }
+                }
+                gst::PadProbeReturn::Ok
+            });
+
```

Comment:
huh? imo this should be handled by the query() virtual  method in the element, not through a probe in ghost pads... dunno... looks odd